eng/_util/cmd/generate-codeql-yml/generate-codeql-yml.go

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
+)
+
+const (
+	patchDir        = "patches"
+	goSubmodulePath = "go"
+	outputFile      = "CodeQL.yml"
+	yamlHeader      = "# This file is auto-generated by generate-codeql-yml"
+	libraryHeader   = "path_classifiers:\n  library:\n    # Classify the upstream go submodule as a 3rd-party library\n    - \"" + goSubmodulePath + "/**\"\n    # Except for these Go source files which we patch"
+)
+
+var (
+	diffRegex      = regexp.MustCompile(`^diff --git a/(\S+)`)
+	goFileSuffix   = ".go"
+	testFileSuffix = "_test.go"
+)
+
+func main() {
+	pathsToExclude := make(map[string]bool)
+
+	err := filepath.Walk(patchDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil || info.IsDir() || !strings.HasSuffix(path, ".patch") {
+			return nil
+		}
+		file, err := os.Open(path)
+		if err != nil {
+			return fmt.Errorf("failed to open patch file %s: %w", path, err)
```

Comment:
The error coming out of `os.Open` already has the operation name and file path, no need to add that context again.

```suggestion
			return err
```